cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableManager.java

```diff
@@ -188,7 +196,8 @@ public QueryHandle enableDataset(Id.DatasetInstance datasetID, DatasetSpecificat
           ((RecordScannable) dataset).getRecordType() : ((RecordWritable) dataset).getRecordType();
 
         // if the type is a structured record, use the schema property to create the table
-        if (StructuredRecord.class.equals(recordType)) {
+        // Use == because that's what same class means.
+        if (StructuredRecord.class == recordType) {
```

Comment:
It was checking for equals, which is the same as "==" (in fact equals check with have a warning).